docs/rfcs/002-archival_storage/README.md

```diff
@@ -290,15 +290,16 @@ Content-Type: application/json
 When storing an update, the service will:
 
 - Check that the supplied version matches the current version
-- Validate and unpack the supplied bag
+- Unpack the supplied bag
 - Store the supplied bag as a new version
-- Update the current version of the bag
+- Register the new version of the bag as the current version
 
-Partial updates, where files that are not changed are not resupplied, are supported through the use of `fetch.txt` in the supplied bag. File references must include the bag version in which the file was previously supplied.
+Partial updates, where files that are not changed are not resupplied, are supported through the use of `fetch.txt` in the supplied bag. File references must specify the full storage location of the previously supplied file, including the version number of the bag in which it was last supplied in the path.
 
 Updates with fetch files should be processed as follows:
 
-- Check that files in `fetch.txt` exist
+- Check that files in `fetch.txt` reference files in the correct bag
+- Check that files in `fetch.txt` exist most recently at the specified version
```

Comment:
We should make this explicit too, does the fetch as described here only support references to files already in the storage service (which seems a sensible first step)?  If so how should those entries be recognised, just by matching bucket/space/bag. If entries do not match a previous version or attempt to skip a version by pointing to an older version what should be done?  If we are match only against the previous bag version that could be explicit here too -- either match a fetch entry in the previous version or an existing file.

Reply:
Is that not covered by "exist most recently at the specified version"? If these checks fail, the ingest should fail.